mage_ai/frontend/components/PipelineDetail/UploadPipeline/index.tsx

```diff
@@ -0,0 +1,175 @@
+import router from 'next/router';
+import { useMemo, useState } from 'react';
+
+import ApiErrorType from '@interfaces/ApiErrorType';
+import Button from '@oracle/elements/Button';
+import FileUploader from '@components/FileUploader';
+import FileType from '@interfaces/FileType';
+import FlexContainer from '@oracle/components/FlexContainer';
+import Panel from '@oracle/components/Panel';
+import PopupMenu from '@oracle/components/PopupMenu';
+import ProgressBar from '@oracle/components/ProgressBar';   
+import Spacing from '@oracle/elements/Spacing';
+import Table from '@components/shared/Table';
+import Text from '@oracle/elements/Text';
+import ToggleSwitch from '@oracle/elements/Inputs/ToggleSwitch';
+import { DropZoneStyle, TableStyle } from './index.style';
+import { isEmptyObject } from '@utils/hash';
+import { sortByKey } from '@utils/array';
+import { UNIT } from '@oracle/styles/units/spacing';
+import { useModal } from '@context/Modal';
+
+type UploadPipelineProps = {
+  fetchPipelines?: () => void;
+  onCancel: () => void;
+};
+
+function UploadPipeline({
+  fetchPipelines,
+  onCancel,
+}: UploadPipelineProps) {
+  const [overwriteVariable, setOverwriteVariable] = useState<boolean>(false);
+  const [isDragActive, setIsDragActive] = useState<boolean>(false);
+  const [fileUploadProgress, setFileUploadProgress] = useState<{
+    [key: string]: number;
+  }>({});
+  const [uploadedPipeline, setUploadedPipeline] = useState<{
+    [key: string]: ApiErrorType & FileType;
+  }>({});
+  const hasFiles: boolean = !isEmptyObject(fileUploadProgress);
+
+  const [showPopupMenu] = useModal(() => {
+    const pipeline: ApiErrorType & FileType = Object.values(uploadedPipeline)[0];
+    const errorMessage = pipeline?.message;
+    const error = !!errorMessage;
+
+    // inverted confirm and cancel buttons because Popup component cannot hide confirm
+    return <PopupMenu
+          centerOnScreen
+          confirmText={'Close'}
+          {...(!error ? { cancelText: `Edit ${pipeline?.name}` } : null)}
```

Comment:
Seems clearer if this was refactored like below:
`cancelText: !error ? `Edit ${pipeline?.name}` : null`

Reply:
Oh right, that's what I meant. Forgot it was a prop on a component so had to update the syntax. Thanks
```
cancelText={!error ? `Edit ${pipeline?.name}` : null}
```